src/transformers/models/codegen/modeling_codegen.py

```diff
@@ -151,7 +151,7 @@ def _attn(
 
         # compute causal mask from causal mask buffer
         query_length, key_length = query.size(-2), key.size(-2)
-        causal_mask = self.causal_mask[:, :, key_length - query_length : key_length, :key_length]
+        causal_mask = self.causal_mask[:, :, key_length - query_length : key_length, :key_length].to(torch.uint8)
```

Comment:
Let's have a comment here to explain why we need `.to(torch.uint8)` 🙏

Reply:
Added more comments on 8b81ac12339692bb3bbddc363ddcb9438eeb3e68 
Let me know if anything is unclear!

Reply:
Stupid question, why is that not
```suggestion
        self.causal_mask = self.causal_mask[:, :, key_length - query_length : key_length, :key_length].to(torch.uint8)
```
feels like something you want to do only once no?

Reply:
It won't be needed anywhere as we found the root cause with Younes :-)